File: Data Structures & Algorithms/is-subsequence/submission-1.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(t.size()<s.size()){
            return false;
        }
        int j = 0;
        for(int i = 0; i<s.size();i++){
                if(j>=t.size()){
                        return false;
                }
                else{
                        while(j<t.size()){
                        if(t[j] == s[i]){
                                j++;
                                break;
                        }
                        j++;
                        }
                }
            
        }
        
        return true;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:


        def dfs(curr):
            if not curr:
                return 0
            
            left = dfs(curr.left)
            if left < 0:
                return -1
            right = dfs(curr.right)
            if right < 0:
                return -1

            if abs(left-right) > 1:
                return -1

            return 1 + max(left,right)


        
        if dfs(root) == -1:
            return False
        else:
            return True